lib/internal/bootstrap/realm.js

```diff
@@ -310,9 +310,20 @@ class BuiltinModule {
   }
 
   static isBuiltin(id) {
-    return BuiltinModule.canBeRequiredWithoutScheme(id) || (
-      typeof id === 'string' &&
-        StringPrototypeStartsWith(id, 'node:') &&
+    if (typeof id !== 'string') {
+      return false;
+    }
+
+    if (StringPrototypeStartsWith(id, '_')) {
```

Comment:
Can you add comments to each if statement to understand the reasoning.

Reply:
sure :+1:

Reply:
is this ok? 6129a75b12eff100823f5a4d32a37411f5c0917f